openjdk/mmtkBarrierSet.hpp

```diff
@@ -35,7 +35,7 @@
 #include "utilities/fakeRttiSupport.hpp"
 #include "mmtk.h"
 
-#define MMTK_ENABLE_ALLOCATION_FASTPATH true
+#define MMTK_ENABLE_ALLOCATION_FASTPATH false
```

Comment:
Turn-off the fast path will also slow down the other collectors (by a lot). We need to figure out a way to only disable fast path for your mallocms gc. Probably we need to refactor this code.

Reply:
I think `MMTK_ENABLE_ALLOCATION_FASTPATH` does not have to be a macro. We can turn this into a global variable (or probably a static field in MMTkHeap) and during VM-initialisation, set this to false only if MallocMS is the selected collector.